libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/service/MapboxTripService.kt

```diff
@@ -79,13 +93,13 @@ internal class MapboxTripService(
 
     private val serviceStarted = AtomicBoolean(false)
 
-    private fun postDataToChannel() {
-        notificationDataChannel.offer(
-            MapboxNotificationData(
-                tripNotification.getNotificationId(),
-                tripNotification.getNotification()
-            )
+    private fun updateNotificationData() {
+        val notificationData = MapboxNotificationData(
+            tripNotification.getNotificationId(),
+            tripNotification.getNotification(),
         )
+        notificationDataObservers.forEach { it.onNotificationUpdated(notificationData) }
+        notificationDataObservers.clear()
```

Comment:
Why are we clearing the observers here? I don't think this should be here.

Reply:
This is done in order to invoke these observers only once, which the name `registerOneTimeNotificationDataObserver` suggests. It doesn't make much sense to let those observers stay longer, because they won't be invoked on every notification update, but only when another observer is added.